openpype/lib/transcoding.py

```diff
@@ -315,6 +315,92 @@ def parse_oiio_xml_output(xml_string, logger=None):
     return output
 
 
+def get_channels_info_by_layer_name(channel_names):
+    """Get channels info grouped by layer name.
+
+    Finds all layers in channel names and returns list of dictionaries with
+    information about channels in layer.
+    Example output (not real world example):
+        [
+            {
+                "name": "Main",
+                "review_channels": {
+                    "R": "Main.red",
+                    "G": "Main.green",
+                    "B": "Main.blue",
+                    "A": None,
+                },
+                "channel_names": {
+                    "Main.red": "red",
+                    "Main.green": "green",
+                    "Main.blue": "blue",
+                    "Main.z": "z"
+                }
+            },
+            {
+                "name": "Composed",
+                "review_channels": {
+                    "R": "Composed.R",
+                    "G": "Composed.G",
+                    "B": "Composed.B",
+                    "A": "Composed.A",
+                },
+                "channel_names": {
+                    "Composed.R": "R",
+                    "Composed.G": "G",
+                    "Composed.B": "B",
+                    "Composed.A": "A",
+                }
+            },
+            ...
+        ]
+
+    Args:
+        channel_names (list[str]): List of channel names.
+
+    Returns:
+        list[dict]: List of channels information.
+    """
+
+    layer_names_order = []
+    rgba_by_layer_name = collections.defaultdict(dict)
+    channels_by_layer_name = collections.defaultdict(dict)
+    for channel_name in channel_names:
+        name_parts = channel_name.split(".")
+        last_part = name_parts.pop(-1)
+        layer_name = ".".join(name_parts)
+        channels_by_layer_name[layer_name][channel_name] = last_part
+        rgb_part = last_part.lower()
+        if layer_name not in layer_names_order:
+            layer_names_order.append(layer_name)
+        if rgb_part in ("r", "red"):
+            rgba_by_layer_name[layer_name]["R"] = channel_name
+        elif rgb_part in ("g", "green"):
+            rgba_by_layer_name[layer_name]["G"] = channel_name
+        elif rgb_part in ("b", "blue"):
+            rgba_by_layer_name[layer_name]["B"] = channel_name
+        elif rgb_part in ("a", "alpha"):
+            rgba_by_layer_name[layer_name]["A"] = channel_name
+
+    if "" in layer_names_order:
+        layer_names_order.remove("")
+        layer_names_order.insert(0, "")
```

Comment:
So what's this? We want to prioritize channels that do not have one of the RGBA suffixes? Note that there could be muliple channels with "" layer name here if inputs are:
```python
channel_names = ["red", "green", "blue", "z", "custom_aov"]
```